File: backend/app/models/market_price.py
"""Daily fish market price reference data, used by the Market Price screen."""
from sqlalchemy import Column, Integer, Float, String, Date, DateTime, func
from app.database import Base


class MarketPrice(Base):
    __tablename__ = "market_prices"

    id = Column(Integer, primary_key=True, index=True)
    species = Column(String(80), nullable=False, index=True)
    market_name = Column(String(120), nullable=False)
    harbor_region = Column(String(120), nullable=False, index=True)
    price_per_kg = Column(Float, nullable=False)
    currency = Column(String(8), nullable=False, default="INR")
    price_date = Column(Date, nullable=False, index=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: mobile/lib/models/market_price.dart
class MarketPriceInfo {
  final int id;
  final String species;
  final String marketName;
  final String harborRegion;
  final double pricePerKg;
  final String currency;
  final String priceDate;

  MarketPriceInfo({
    required this.id,
    required this.species,
    required this.marketName,
    required this.harborRegion,
    required this.pricePerKg,
    required this.currency,
    required this.priceDate,
  });

  factory MarketPriceInfo.fromJson(Map<String, dynamic> json) => MarketPriceInfo(
        id: json['id'] as int,
        species: json['species'] as String,
        marketName: json['market_name'] as String,
        harborRegion: json['harbor_region'] as String,
        pricePerKg: (json['price_per_kg'] as num).toDouble(),
        currency: json['currency'] as String,
        priceDate: json['price_date'] as String,
      );

  Map<String, dynamic> toCacheJson() => {
        'id': id,
        'species': species,
        'market_name': marketName,
        'harbor_region': harborRegion,
        'price_per_kg': pricePerKg,
        'currency': currency,
        'price_date': priceDate,
      };
}
